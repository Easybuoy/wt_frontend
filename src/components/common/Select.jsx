import React from "react";
import { Select as ChakraSelect } from "@chakra-ui/core";
import styled from "styled-components";
import PropTypes from "prop-types";
const CustomStyledSelect = styled.div`
  margin: 1rem 0;
  width: 100%;

  select {
    display: block;
    font-size: 16px;
    outline: none;
    color: #a3b1c2;
    line-height: 1.3;
    box-sizing: border-box;
    -webkit-appearance: none;
    background-color: #fffcf2;
    background-size: 3em, 100%;
    height: auto;
    padding: 1.3rem;
    border-style: solid;
    border-radius: 0;
    background-repeat: no-repeat, repeat;
    border: 1.5px solid rgba(0, 0, 0, 0.25);
  }
  span {
    color: red;
  }
`;

const Select = ({ options, name, onChange, placeholder, error, value }) => {
  return (
    <CustomStyledSelect>
      <ChakraSelect
        name={name}
        onChange={onChange}
        className="dropdown"
        placeholder={placeholder}
        value={value}
        bg="#FFFCF2"
        _hover="black"
        focusBorderColor="#FF8744"
        errorBorderColor="crimson"
      >
        {options.map(option => (
          <option key={option.text} value={option.value}>
            {option.text}
          </option>
        ))}
      </ChakraSelect>
      <span>{error}</span>
    </CustomStyledSelect>
  );
};

export default Select;

Select.propTypes = {
  options: PropTypes.array.isRequired,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  placeholder: PropTypes.string.isRequired,
  error: PropTypes.string,
  value: PropTypes.string
};

Select.defaultProps = {
  options: []
};
