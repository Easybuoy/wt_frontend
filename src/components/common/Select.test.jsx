import React from "react";
import { shallow } from "enzyme";

import Select from "./Select";

describe("<Select />", () => {
  let wrapper;

  const props = {
    options: [],
    name: "",
    onChange: () => {},
    placeholder: ""
  };

  beforeEach(() => {
    wrapper = shallow(<Select {...props} />);
  });

  it("renders the Preview component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
