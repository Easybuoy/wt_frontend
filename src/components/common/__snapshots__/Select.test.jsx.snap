// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> renders the Preview component correctly 1`] = `
<styled.div>
  <Select
    _hover="black"
    bg="#FFFCF2"
    className="dropdown"
    errorBorderColor="crimson"
    focusBorderColor="#FF8744"
    name=""
    onChange={[Function]}
    placeholder=""
  />
  <span />
</styled.div>
`;
